src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Extensions/RazorCodeDocumentExtensions.cs

```diff
@@ -54,7 +53,7 @@ public static SyntaxTree GetOrParseCSharpSyntaxTree(this RazorCodeDocument docum
 
     public static bool TryGetGeneratedDocument(
         this RazorCodeDocument codeDocument,
-        Uri generatedDocumentUri,
+        DocumentUri generatedDocumentUri,
```

Comment:
It's not necessarily a huge problem, but this is a good example of a thought I'd had a few times looking at the changes: Are we going a little bit too far with them?

The idea is to get Uri parsing out of LSP message deserialization, so moving away from Uri to DocumentUri for everything that is at the boundary makes sense. This feels like a step further, where we're essentially saying that we'll be able to possibly produce a `RazorCodeDocument` even if a Uri is not valid, and that doesn't feel right. It feels like for this API, and for others at a similar level, we are probably fine to just expect the LSP message handlers to pass in the parsed Uri, so that if there is invalid data coming in it's validated relatively early (just not as early as deserialization).

Reply:
> even if a Uri is not valid

Wanted to make a clarification that these are generally actual valid URIs - but `System.Uri` doesn't handle them correctly.  So ultimately that means these URIs are not parse-able (unless we write our own parser).

Generally it is up to you if parsing the URI is a requirement for running your handlers.  For Roslyn it isn't a hard requirement (they'll become misc files)

Reply:
Oooh thats a good distinction I missed/forgot. We definitely use the filename of the Uri to find/create various documents, so I think technically we do need things to be parsable. Probably a good rule to use though (ie, `IFilePathService` is the thing that does that work, so it should take `Uri` and let that inform the changes). Seems like we could go either way though really, and things would be pretty viral either way. Doesn't feel like something to nail down in this PR, but we can pivot later as we work out what works best.